src/app/applyContextMenu.js

```diff
@@ -0,0 +1,113 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { clipboard, Menu, MenuItem } from 'electron'
+
+/**
+ * Enable context menu by right click
+ *
+ * @param {import('electron').BrowserWindow} browserWindow - Browser window
+ */
+export function applyContextMenu(browserWindow) {
+	browserWindow.webContents.on('context-menu', async (event, params) => {
+		const menu = new Menu()
+
+		// Add context actions for misspeling words and typos
+		const menuMisspellingItems = [
+			...params.dictionarySuggestions.map(suggestion => ({
+				label: suggestion,
+				click: () => browserWindow.webContents.replaceMisspelling(suggestion),
+			})),
+			{
+				label: 'Add to dictionary',
+				click: () => browserWindow.webContents.session.addWordToSpellCheckerDictionary(params.misspelledWord),
+			},
+			{ type: 'separator' },
+		]
+		if (params.misspelledWord) {
+			menuMisspellingItems.forEach(item => menu.append(new MenuItem(item)))
+		}
+
+		// Add context actions for handling images
+		const menuImageItems = [
+			{
+				label: 'Copy image',
+				click: () => browserWindow.webContents.copyImageAt(params.x, params.y),
+			},
+			{
+				label: 'Copy image link address',
+				click: () => clipboard.writeText(params.srcURL),
+			},
+			{ type: 'separator' },
+		]
+
+		// Add context actions for handling links
+		const menuLinkItems = [
+			{
+				label: 'Copy link address',
+				click: () => clipboard.writeText(params.linkURL),
+			},
+			{
+				label: 'Copy link text',
+				click: () => clipboard.writeText(params.linkText),
+			},
+			{ type: 'separator' },
+		]
+
+		// Add context actions for clipboard events and text editing
+		const menuClipboardItems = [
+			{
+				role: 'copy',
+				enabled: params.selectionText && params.editFlags.canCopy,
+			},
+			{
+				role: 'cut',
+				enabled: params.selectionText && params.editFlags.canCut,
+			},
+			{
+				role: 'selectAll',
+				enabled: params.editFlags.canSelectAll,
+			},
+			{
+				role: 'paste',
+				enabled: params.editFlags.canPaste,
+			},
+			{ type: 'separator' },
+		]
+
+		// Show actions depending on context
+		if (params.srcUrl || params.hasImageContents) {
+			menuImageItems.forEach(item => menu.append(new MenuItem(item)))
+		} else if (params.linkUrl || params.linkText) {
+			menuLinkItems.forEach(item => menu.append(new MenuItem(item)))
```

Comment:
All "internal" links are also caught here. It's links like:
- `file://path/to/NextcloudTalk/bla/bla/talk_window/index.html#/call/123123` on production
- `http://localhost:3000/talk_window/index.html#/call/123123` on development

These are definitely not the links we want users to copy.

We can handle those links with `isExternalLink` helper in 
https://github.com/nextcloud/talk-desktop/blob/main/src/app/externalLinkHandlers.js#L41